lib/puppet_x/bsd/hostname_if/inet.rb

```diff
@@ -34,9 +37,24 @@ def initialize(addresses)
 
         def process
           @addrs.each {|a|
-            # Return the dynamic address assignemnt if found
-            if a =~ /^(rtsol|dhcp)$/
+            # Return the dynamic address assignment if found
+            if a =~ /^(dhcp)$/
               yield a
+	    elsif a =~ /^(rtsol|inet6 autoconf)$/
```

Comment:
In this block and the block above, when testing for `rtsol` or  `inet6 autoconf`, it seems that something like the following would have the same logic.

```Puppet
elsif a =~ /^(rtsol|inet6 autoconf)$/
  kernelversion = Facter.value('kernelversion')
  if versioncmp(kernelversion, "5.6") < 0
    yield 'rtsol'
  else
    yield 'inet6 autoconf'
  end
end
```

Unless I'm overlooking something, I think that still captures the same idea.